browser/models.py

```diff
@@ -38,6 +41,31 @@ def compile_pickle_mysql(type_, compiler, **kw):
     return 'LONGBLOB'
 
 
+class MutableNdarray(Mutable, np.ndarray):
```

Comment:
Can you help me understand what this does/what problem it solves?

Reply:
When we change a built-in column type in SQLAlchemy (such as Integer or String), SQLAlchemy knows when the column has been changed and commits the change to the database the next time we call db.session.commit(). 

When using PickleType columns, SQLAlchemy does not have any insight into the mutability/changes to the object it is pickling, and instead only checks the object by reference. When we mutate a numpy array, for instance when replacing a frame with new edited data with `self.frame[..., self.feature] = img`, SQLAlchemy would normally see this object as unchanged as the reference to the array has not changed.

With the Mutable mixin, we can modify the behavior of functions that mutate a numpy array, like`__setitem__`, so that it calls `self.changed()` to notify SQLAlchemy that the data has been changed and needs to be written to the database upon the next commit.

We have not yet used the Mutable mixin with the Labels table, which has two PickleType columns for `cell_ids` and `cell_info`. Therefore, for changes to these columns to be persisted, we copy the dictionaries and reassign them so that the reference has changed and SQLAlchemy is guaranteed to save the changes. There are some TODO comments that show where lack of Mutable for cell_ids/cell_info, and we should aim to use the Mutable mixin for these columns as well so that SQLAlchemy monitors and manages the changes in the background, replacing the code that explicitly copies the data before committing.

Reply:
Just to confirm, would adding the Mutable mixin to the other columns in the future require another db schema change?

Reply:
I don't think it would require a schema change as the Mutable mixin is just on the ORM side and wouldn't change the column declarations on the SQL side. Still, if it does require a schema change, I'd imagine it would be a straightforward migration as we wouldn't be adding or moving columns, just changing how these columns behave.